block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -110,27 +109,26 @@ retrievalWorkerImpl keepAliveTimer SendActions {..} =
         slotDuration <- fromIntegral . toMicroseconds <$> getCurrentEpochSlotDuration
         setTimerDuration keepAliveTimer $ 3 * slotDuration
         startTimer keepAliveTimer
+        -- Exception handlers are installed locally, on the 'thingToDoNext',
+        -- to ensure that network troubles, for instance, do not kill the
+        -- worker.
         thingToDoNext
```

Comment:
I think any other exceptions should be considered fatal. `classifyNewHeader` and `getProcessBlocks` are the pieces that need exception handling. Both of them may throw exceptions: `classifyNewHeader` uses a `runExceptT` out front, but inside there's a lot of effectful stuff like `DB.getTipHeader`, `GS.getAdoptedBVFull`, `LrcDB.getLeadersForEpoch`, etc. so I believe it could throw IO exceptions.